keps/sig-apps/4026-job-creation-timestamp-annotation/README.md

```diff
@@ -0,0 +1,299 @@
+
+The CronJob controller will begin adding the scheduled timestamp as an annotation to jobs created while the feature is enabled, and existing jobs will be unaffected.
+
+###### Are there any tests for feature enablement/disablement?
+
+We plan to add unit tests.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+It will not impact already running workloads.
+
+###### What specific metrics should inform a rollback?
+
+- Users can monitor queue related metrics (e.g., queue depth and work duration) to make sure they aren't growing.
+- For CronJobs, users can also monitor `job_creation_skew_duration_seconds`.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+The feature will be tested manually prior to beta launch.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+N/A
+
+###### How can an operator determine if the feature is in use by workloads?
+
+- Check if CronJobs have the annotation `batch.kubernetes.io/cronjob-scheduled-timestamp`.
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [ ] Events
+  - Event Reason:
+- [X] API .metadata
+  - Condition name:
+  - Other field:
+    - `.metadata.annotations['batch.kubernetes.io/cronjob-scheduled-timestamp']`
+- [ ] Other (treat as last resort)
+  - Details:
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+- 99% percentile over day for Job syncs is <= 15s for a client-side 50 QPS limit.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [X] Metrics
+  - Metric name:
+  - `job_creation_skew_duration_seconds`.
+  - [Optional] Aggregation method:
+  - Components exposing the metric:
+- [ ] Other (treat as last resort)
+  - Details:
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+No.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+New job annotation of size 34B plus value of size N where N is the number of digits in the job ordinal.
+Worst case for N would be the max number of jobs per node * max number of nodes.
+Per the docs on [large clusters](https://kubernetes.io/docs/setup/best-practices/cluster-large/), this would be 110 pods/node * 5000 nodes = 550,000 (6 digits).
```

Comment:
FWIW - we support 150,000 pods:
https://github.com/kubernetes/community/blob/master/sig-scalability/configs-and-limits/thresholds.md

But I have no idea how this is related...

Reply:
Removed unneeded statement.